### PR TITLE
Benchmark: Adds a workload to insert and read the inserted item.

### PR DESCRIPTION
**Pull request by j82w**
# Pull Request Template

## Description

Adds a new benchmark to simulate users inserting a document and immediately reading it. This with the https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4478 will test replication latency between different geo regions, and how SDK handles various error scenarios.

## Type of change

Please delete options that are not relevant.

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] This change requires a documentation update

## Closing issues

To automatically close an issue: closes #IssueNumber

### CODE REVIEW

**Review by RaviTella — APPROVED** [review]
LGTM. Thanks Jake.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        input,
+                        partitionKey);
+
+                ruCharges = itemResponse.Headers.RequestCharge;
+                writeDiagnostics = itemResponse.Diagnostics;
+                System.Buffers.ArrayPool<byte>.Shared.Return(input.GetBuffer());
+                
+                if (!itemResponse.IsSuccessStatusCode)
+                {
+                    return new OperationResult()
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs:61]: Might be an existing gap already for other benchmarks. 

TelemetrySpan is not failures are not reflected in-case of non-exceptions. Can be an independent follow-up.

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            CosmosDiagnostics writeDiagnostics = null;
+            using (MemoryStream input = JsonHelper.ToStream(this.sampleJObject))
+            {
+                ResponseMessage itemResponse = await this.container.CreateItemStreamAsync(
+                        input,
+                        partitionKey);
+
+                ruCharges = itemResponse.Headers.RequestCharge;
+                writeDiagnostics = itemResponse.Diagnostics;
+                System.Buffers.ArrayPool<byte>.Shared.Return(input.GetBuffer());
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs:57]: Good catch, existing usages are not releasing it.
@j82w can you please create a follow-up to address the gaps (may be wrapped MemoryStream might be a solution)

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs (hunk 1 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs:61]: @NaluTripician can you please take a look?

**Review by kirankumarkolli — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs (hunk 1 above)
Comment (NaluTripician) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs:61]: Issue for tracking https://github.com/Azure/azure-cosmos-dotnet-v3/issues/4524

File: Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs (hunk 2 above)
Comment (NaluTripician) [reply Microsoft.Azure.Cosmos.Samples/Tools/Benchmark/v3/InsertReadV3BenchmarkOperation.cs:57]: Issue for tracking https://github.com/Azure/azure-cosmos-dotnet-v3/issues/4525

**Pull request merged**